### PR TITLE
Prepare for building with rebar3

### PR DESCRIPTION
**Pull request by raghavkarol**
Removed dependency on `gproc` and `erlang.mk`. Chronos now builds with rebar2 and rebar3 out of the box

### CODE REVIEW

File: src/chronos.erl [hunk 1]
```diff
@@ -78,9 +78,8 @@ stop_timer(ServerName, TimerName) ->
...
 
 
--spec init(server_name()) -> {'ok', #chronos_state{}} | {'stop', term()} | 'ignore'.
-init(ServerName) ->
-    gproc:reg(proc_name(ServerName)),
+-spec init([]) -> {'ok', #chronos_state{}} | {'stop', term()} | 'ignore'.
+init(_Args) ->
     {ok, #chronos_state{}}.
 
 -spec handle_call(term(), term(), #chronos_state{}) ->
```
Comment (elvisci) [inline src/chronos.erl]: The spec in line 85 uses a record, please define a type for the record and use that instead.

File: src/chronos.erl [hunk 2]
```diff
@@ -78,9 +78,8 @@ stop_timer(ServerName, TimerName) ->
 %%%===================================================================
 
 
--spec init(server_name()) -> {'ok', #chronos_state{}} | {'stop', term()} | 'ignore'.
-init(ServerName) ->
-    gproc:reg(proc_name(ServerName)),
+-spec init([]) -> {'ok', #chronos_state{}} | {'stop', term()} | 'ignore'.
```
Comment (elvisci) [inline src/chronos.erl]: The spec in line 81 uses a record, please define a type for the record and use that instead.

File: src/chronos.erl [hunk 3]
```diff
@@ -51,7 +51,7 @@
 
 -spec start_link(server_name()) -> {'ok', pid()} | 'ignore' | {'error',term()}.
```
Comment (elvisci) [inline src/chronos.erl]: Missing space right "," on line 52

File: src/chronos.erl [hunk 4]
```diff
@@ -78,12 +80,11 @@ stop_timer(ServerName, TimerName) ->
...
--spec init(server_name()) -> {'ok', #chronos_state{}} | {'stop', term()} | 'ignore'.
-init(ServerName) ->
-    gproc:reg(proc_name(ServerName)),
+-spec init([]) -> {'ok', state()} | {'stop', term()} | 'ignore'.
+init(_Args) ->
     {ok, #chronos_state{}}.
 
--spec handle_call(term(), term(), #chronos_state{}) ->
+-spec handle_call(term(), term(), state()) ->
                          {'reply', 'ok' | {'error', term()} , #chronos_state{}} |
```
Comment (elvisci) [inline src/chronos.erl]: Line 88 is too long:                          {'reply', 'ok' | {'error', term()} , #chronos_state{}} |.

File: src/chronos.erl [hunk 5]
```diff
@@ -78,12 +80,11 @@ stop_timer(ServerName, TimerName) ->
...
 
--spec init(server_name()) -> {'ok', #chronos_state{}} | {'stop', term()} | 'ignore'.
-init(ServerName) ->
-    gproc:reg(proc_name(ServerName)),
+-spec init([]) -> {'ok', state()} | {'stop', term()} | 'ignore'.
+init(_Args) ->
     {ok, #chronos_state{}}.
 
--spec handle_call(term(), term(), #chronos_state{}) ->
+-spec handle_call(term(), term(), state()) ->
```
Comment (elvisci) [inline src/chronos.erl]: The spec in line 87 uses a record, please define a type for the record and use that instead.

**Comment by lehoff** [comment]
Very good change.

**Pull request merged**